http/src/v0/root_files/add.rs

```diff
@@ -408,15 +404,37 @@ mod tests {
         );
     }
 
-    async fn tokio_ipfs() -> ipfs::Ipfs<ipfs::TestTypes> {
-        let options = ipfs::IpfsOptions::inmemory_with_generated_keys();
-        let (ipfs, fut) = ipfs::UninitializedIpfs::new(options, None)
-            .await
-            .start()
-            .await
-            .unwrap();
+    // copied from js-ipfs/packages/ipfs/test/http-api/inject/files.js
+    #[tokio::test(max_threads = 1)]
+    async fn js_multipart_test() {
+        let ipfs = Node::new("0").await;
+        let mut rng = rand::thread_rng();
+
+        let response = warp::test::request()
+            .path("/add")
+            .header(
+                "Content-Type",
+                "multipart/form-data; boundary=----------287032381131322",
+            )
+            .body(
+                format!(
+                    "\r\n\
+                    ------------287032381131322\r\n\
+                    Content-Disposition: form-data; name=\"test\"; filename=\"test.txt\"\r\n\
+                    Content-Type: text/plain\r\n\
+                    \r\n\
+                    {}\r\n\
+                    ------------287032381131322--",
+                    iter::repeat(())
+                        .map(|_| rng.sample(Alphanumeric))
+                        .take(1024 * 1024 * 2)
+                        .collect::<String>()
+                )
+                .into_bytes(),
+            )
+            .reply(&add(&ipfs))
+            .await;
 
-        tokio::spawn(fut);
-        ipfs
+        assert_eq!(response.status(), 200);
```

Comment:
is this test case is just to accept the text/plain as I guess js-ipfs accepts it?

accepting text/plain sounds really strange be honest ... wouldn't that require some charset wrangling as well? I guess there's no issues if you assume ascii or utf8 but ... Why?

Reply:
this was the way `js-ipfs` did it, I just followed suit; since it's random alphanumeric bytes there's no encoding issues